src/fluid.rs

```diff
@@ -0,0 +1,434 @@
+extern crate image;
+extern crate rand;
+
+use geo::algorithm::rotate::RotatePoint;
+use geo::{line_string, point};
+use noise::{NoiseFn, Perlin};
+use rand::Rng;
+
+//#[derive(Copy, Clone)]
+pub struct SimulationConfig {
+    pub dt: f32,
+    pub iter: u32,
+    pub scale: u32,
+    pub t: f32,
+    pub size: u32,
+}
+
+impl Default for SimulationConfig {
+    fn default() -> SimulationConfig {
+        SimulationConfig {
+            dt: 0.02,
+            iter: 16,
+            scale: 4,
+            t: 0f32,
+            size: 128,
+        }
+    }
+}
+
+//#[derive(Copy, Clone)]
+pub struct FluidConfig {
+    pub diffusion: f32,
+    pub viscousity: f32,
+}
+
+impl Default for FluidConfig {
+    fn default() -> FluidConfig {
+        FluidConfig {
+            diffusion: 0f32,
+            viscousity: 0.0000001,
+        }
+    }
+}
+
+//#[derive(Copy, Clone)]
+pub struct Fluid {
+    pub fluid_configs: FluidConfig,
+    pub simulation_configs: SimulationConfig,
+    s: Vec<f32>,
+    density: Vec<f32>,
+    velocities_x: Vec<f32>,
+    velocities_y: Vec<f32>,
+    velocities_x0: Vec<f32>,
+    velocities_y0: Vec<f32>,
+}
+
+impl Fluid {
+    pub fn new(init_fluid: FluidConfig, init_simulation: SimulationConfig) -> Fluid {
+        Fluid {
+            s: vec![0.0; (init_simulation.size * init_simulation.size) as usize],
```

Comment:
refactor `init_simulation.size * init_simulation.size as usize` outside as a variable and use it instead of repeating

Reply:
Done